modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/ParameterModelsLoaderDelegate.java

```diff
@@ -98,7 +99,7 @@ public ParameterModelsLoaderDelegate(List<ParameterDeclarerContributor> contribu
                                          ParameterDeclarationContext declarationContext,
                                          ParameterGroupDeclarer parameterGroupDeclarer) {
     List<ParameterDeclarer> declarerList = new ArrayList<>();
-    checkAnnotationsNotUsedMoreThanOnce(parameters, Connection.class, Config.class, MetadataKeyId.class);
+    checkAnnotationsNotUsedMoreThanOnce(parameters, ComponentId.class, Connection.class, Config.class, MetadataKeyId.class);
```

Comment:
This seems to be checked in ParameterModelValidator, I think it is better if it keeps failing there.